packages/viewer/src/components/viewer-pin-tool/viewer-pin-tool.tsx

```diff
@@ -275,4 +297,22 @@ export class ViewerPinTool {
     const rect = getMarkupBoundingClientRect(this.hostEl);
     this.elementBounds = rect;
   }
+
+  private setDepthBuffers(): void {
+    if (
+      this.pins.length > 0 &&
+      this.viewer != null &&
+      this.viewer.depthBuffers == null
+    ) {
+      this.stateMap.shouldClearDepthBuffers = true;
+      this.viewer.depthBuffers = 'final';
+    }
+  }
+
+  private resetDepthBuffers(): void {
+    if (this.stateMap.shouldClearDepthBuffers && this.viewer != null) {
+      this.viewer.depthBuffers = undefined;
+      this.stateMap.shouldClearDepthBuffers = undefined;
+    }
+  }
```

Comment:
Same thought here, could we add a test covering these cases?

Reply:
Added!